kyma-project/busola
# Namespace preselection

**Description**

Changes proposed in this pull request:

- Navigate to NS details if user has a NS in kubeconfig's context
- Get permissions in namespace context instead of cluster context ('*')
- Hide "Back to namespaces" if user has no rights for that

Testing:

1) OIDC way

- Apply following ClusterRole and add a binding for it - bind it to an OIDC group (there's already `some-ns-only` in [OIDC admin](https://apskyxzcl.accounts400.ondemand.com/admin/#/groups/6017c7aebd4ca8455f8a6d18/details).
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: some-ns-only
rules:
- apiGroups:
  - ""
  resourceNames:
  - default
  resources:
  - namespaces
  verbs:
  - get
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - '*'
```
- In enkode, add a kubeconfig without a token, setup OIDC for `IAS400` and add namespace name in a `kubeconfig.contexts[?].context` of your choice.

2) service account way

- Export a kubeconfig in your terminal (`kmain`'s will do)
- Run a [script to generate kk with service account token](https://github.tools.sap/hasselhoff/tools/blob/master/gardener/createServiceAccount.sh)
- (this script assumes `anonymization` namespace - make sure it's on your cluster)
- Use kubeconfig from `generated/` directory in Busola.

Considerations:
- User with restricted permissions still can get back to namespaces (they'll see 403) or open namespace switcher (they'll get a appropriate message). We're fine with that - they'll still have the original link, leading to "their" namespace.
- Users with '*' permissions will still be redirected to the specific namespace if it's set in their kubeconfig - no problems here.

**Related issue(s)**
